### PR TITLE
feat: Add feature switch component

### PR DESCRIPTION
**Pull request by darryltec**
### Added

- Add `FeatureSwitch` component to handle switching of features
image

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/FeatureSwitch/FeatureSwitch.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          <Switch
+            value={enabled}
+            onChange={handleSwitch}
+            ariaLabel={description}
+          />
+          <AnimatePresence>
+            {hasSaveIndicator && savedIndicator && (
+              <motion.div
+                className={styles.savedIndicator}
+                initial={{ y: -4, opacity: 0 }}
```
Comment (darryltec) [inline packages/components/src/FeatureSwitch/FeatureSwitch.tsx]: Gonna extract this out on its own `Animate` component https://github.com/GetJobber/atlantis/pull/126/commits/b586aa7ac8d8d6460980b9ec5b9a01f912be0c1f

**Comment by agglasaur** [comment]
Does the feature content show up before toggling the feature on? Does it make sense to grey out that section if it is? Not sure where we're landing on the hide vs show for that kind of stuff (or if it should be the same as our rules around disabled actions)

**Comment by darryltec** [comment]
> Does the feature content show up before toggling the feature on? 

Yep. Continuing with the "_not knowing what the setting would do until you turn it on_". It would be really nice to have a stricter rule on when to show the feature content our settings and not just a generic case-by-case basis.

> Does it make sense to grey out that section if it is?

Quite possibly. It's gonna be a little harder to do that in this component because you can pass in any children you want inside this component. `opacity` would do it, but yeh...

> Not sure where we're landing on the hide vs show for that kind of stuff (or if it should be the same as our rules around disabled actions)

Same. Sounds like you(@agglasaur ), me, and @ChristyCampos needs to have a lil chat?

**Comment by agglasaur** [comment]
Maybe something we can discuss in design review? Back to the ol patterns
discussion

**Pull request merged**